docker/Dockerfile.neuron.dev

```diff
@@ -1,13 +1,18 @@
+## THIS DOCKERFILE HAS BEEN DEPRECATED ##
```

Comment:
I'd rather we delete this file and leverage another container to describe neuron instructions. We could make neuron an argument of either `Dockerfile` or `Dockerfile.dev`. We shouldn't expect people to read comments in a Dockerfile